tests/Feature/UserControllerTest.php

```diff
@@ -0,0 +1,569 @@
+                'address' 	=> $request_array['address'],
+            ]);
+        $update_response->assertJson( $base_response );
+	}
+
+	public function testUpdateUserFailConditionAgeWrongFormat()
+	{
+		$Response 		= new Response();
+    	$base_response	= $Response->response( '0005', Lang::get('message.web.error.0005'), "", false );
+		$request_array = [
+            'name' 		=> 'abc',
+            'address' 	=> 'Test Dia Chi',
+            'age' 		=> 22,
+        ];
+        $update_request_array = [
+            'id' 		=> 1,
+            'name' 		=> 'def',
+            'address' 	=> 'Update test Dia Chi',
+            'age' 		=> 'abc',
+        ];
+        $response 			= $this->call('POST', 'user/add', $request_array);
+        $update_response 	= $this->call('POST', 'user/edit', $update_request_array);
+        
+        $this->assertDatabaseMissing('users',
+            [
+                'id' 		=> $update_request_array['id'],
+                'name' 		=> $update_request_array['name'],
+                'age' 		=> $update_request_array['age'],
+                'address' 	=> $update_request_array['address'],
+            ]);
+        $this->assertDatabaseHas('users',
+            [
+                'name' 		=> $request_array['name'],
+                'age' 		=> $request_array['age'],
+                'address' 	=> $request_array['address'],
+            ]);
+        $update_response->assertJson( $base_response );
+	}
+
+	public function testDeleteUserSuccess()
+	{
+		$request_array = [
+            'name' 		=> 'abc',
+            'address' 	=> 'Test Dia Chi',
+            'age' 		=> 22,
+        ];
+        $delete_request_array = [
+            'id' 		=> 1,
+        ];
+        $Response 			= new Response();
+    	$base_response		= $Response->response( '200', Lang::get('message.web.success.0003'), $delete_request_array, true );
+        $response 			= $this->call('POST', 'user/add', $request_array);
+        $delete_response 	= $this->call('POST', 'user/delete', $delete_request_array);
+
+        $this->assertDatabaseMissing('users',
+            [
+                'name' 			=> $request_array['name'],
+                'age' 			=> $request_array['age'],
+                'address' 		=> $request_array['address'],
+                'deleted_time' 	=> null
+            ]);
+        $delete_response->assertJson( $base_response );
+	}
+
+	public function testDeleteUserFailConditionIdNotExists()
+	{
+		$request_array = [
+            'name' 		=> 'abc',
+            'address' 	=> 'Test Dia Chi',
+            'age' 		=> 22,
+        ];
+        $delete_request_array = [
+            'id' 		=> 2,
+        ];
+        $Response 			= new Response();
+    	$base_response		= $Response->response( '0001', Lang::get('message.web.error.0001'), "", false );
+        $response 			= $this->call('POST', 'user/add', $request_array);
+        $delete_response 	= $this->call('POST', 'user/delete', $delete_request_array);
```

Comment:
Should assert is there any record that has id = 2 in database before deleting

Reply:
added field id in assert is there any record has id = 2 before deleting